cppapi/server/multiattribute.cpp

```diff
@@ -1756,7 +1840,28 @@ void MultiAttribute::add_alarmed_quality_factor(string &status)
 			status = status + str;
 		}
 	}
+}
 
+//+------------------------------------------------------------------------------------------------------------------
+//
+// method :
+//		MultiAttribute::add_attr()
+//
+// description :
+//		Add given attribute to the attribute list vector and map
+//
+// argument:
+//		in :
+//			- att : The newly configured attribute
+//
+//-------------------------------------------------------------------------------------------------------------------
+void MultiAttribute::add_attr(Attribute *att)
```

Comment:
I think this can be overloaded with `add_attr(Attribute *att, size_t index)` and used to prevent code duplications above.

Reply:
After a little bit more thinking I think it would be better to split this method into two:

`size_t add_attr(Attribute)`, where size_t is an index of the newly added attribute, may be passed in overloaded method

`void put_attr_into_map(Attribute, index)`

And finally aggregate these two in a third one:

```
size_t ndx = add_attr(attr);
put_attr_into_map(attr, ndx)
```

Reply:
I added put_attribute_in_map() method to reduce code duplication... 
add_attr() method is used in some other parts of the code and already existed before this PR so I think we should stick with the same interface for this method.
I could rename put_attribute_in_map in put_attr_into_map if you prefer...

Reply:
@bourtemb that is just fine keep it as it is now!